File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-12.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        m=[0]*26
        i=j=ans=max_fre=0
        while j<len(s):
            m[ord(s[j])-ord('A')]+=1
            max_fre=max(max_fre,m[ord(s[j])-ord('A')])
            while j-i+1-max_fre>k:
                m[ord(s[i])-ord('A')]-=1
                i+=1
            ans=max(ans,j-i+1)
            j+=1
        return ans

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-12.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int map[]=new int[128];
        int i=0,j=0,len=0;
        while(j<s.length()){
            while(map[s.charAt(j)]>0){
                map[s.charAt(i)]--;
                i++;
            }
            len = Math.max(len,j-i+1);
            map[s.charAt(j)]++;
            j++;
        }
        return len;
    }
}
